ytypes/validate.go

```diff
@@ -22,6 +22,26 @@ import (
 	"github.com/openconfig/ygot/ygot"
 )
 
+// ValidationConfig defines options relating to validation against the YANG
+// schema.
+type ValidationConfig struct {
+	// IgnoreMissingLeafrefData determines whether leafrefs that target a node
+	// that does not exist should return an error to the calling application. When
+	// set to true, no error is returned.
+	//
+	// This functionality is typically used where a partial set of schema information
+	// is populated, but validation is required - for example, configuration for
+	// a protocol within OpenConfig references an interface, but the schema being
+	// validated does not contain the interface definitions.
+	IgnoreMissingLeafrefData bool
+}
+
+var (
+	// Config is used to control the behaviour of the Validate functionality
+	// called against a generated struct.
+	Config *ValidationConfig
```

Comment:
From past experience, setting config through a global var may lead to trouble down the road. I recommend passing config around explicitly to Validate rather than setting the global var (even though it will clutter up the Validate API). It is also better to expose this to the users in this manner. I can take this work item if the changes prove to be extensive.

Reply:
Yes, I pondered the same. I wonder whether there's a good way to figure out how to make this API stable going forward - it seems like some interface like gRPC uses might be reasonable:

```golang
type ValidationOption interface {
  IsValidationOption()
}

type LeafrefOptions struct {
  DataMustExist bool
}
func (LeafrefOptions) IsValidationOption() { }

func Validate(opts ...ValidationOption) util.Error { ... }
```

Reply:
sgtm